recipe/gen_patch_json.py

```diff
@@ -504,6 +504,18 @@ def _gen_new_index(repodata, subdir):
                 else:
                     record['constrains'] = ["typing_extensions"]
 
+        if record_name == "ipython" and record.get('timestamp', 0) < 1609621539000:
+            # https://github.com/conda-forge/ipython-feedstock/issues/127
+            if record["version"] in ["7.18.0"]:
+                i = record['depends'].index('jedi >=0.16')
+                record['depends'][i] = 'jedi >=0.10,<0.18'
+            else:
+                # before conda-forge/ipython-feedstock#110 (7.18.0), this has been 0.10 for years, see
+                # https://github.com/conda-forge/ipython-feedstock/blame/a487716d78710c4f8998654c63000cb5c3402910/recipe/meta.yaml#L30
+                # and returned to that with conda-forge/ipython-feedstock#111 (7.18.1)
+                i = record['depends'].index('jedi >=0.10')
+                record['depends'][i] = 'jedi >=0.10,<0.18'
+
```

Comment:
These code blocks need to figure out what heppens if the string you are looking at is not in the depends list. Also, see some of the functions in this file. They may be able to be used to do this edit.